.github/workflows/build-preview-environment.yml

```
@@ -0,0 +1,108 @@
+        id: sanitize_env
+        shell: bash
+        run: |
+          SANITIZED_BRANCH_NAME=$(echo -n ${{ env.REF }} | tr "/" "-")
+          echo "Sanitized branch name: $SANITIZED_BRANCH_NAME"
+          TRIMMED_BRANCH_NAME=$(echo -n "$SANITIZED_BRANCH_NAME" | cut -c 1-8)
+          echo "sanitized_env_name=$SANITIZED_BRANCH_NAME" >> $GITHUB_OUTPUT;
+          echo "trimmed_env_name=$TRIMMED_BRANCH_NAME" >> $GITHUB_OUTPUT;
+        
+      - name: Environment deployment
+        id: env-name
+        run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2086:info:2:74: Double quote to prevent globbing and word splitting

(shellcheck)